DIFM: Add descriptions and badges on Page Picker step

aneeshd16 opened the pull request:
#### Proposed Changes

Ref: pdh1Xd-1sI-p2

* Adds page descriptions as tooltips on the page picker step in the DIFM flow.

https://user-images.githubusercontent.com/5436027/195095073-999915e3-db2a-41ad-befd-faac8fe0a683.mp4

* Adds the Popular badge to the Services, Pricing and FAQ pages.

<img width="772" alt="image" src="https://user-images.githubusercontent.com/5436027/195361777-5de6c1a4-6915-432b-bf6e-386662f50593.png">


#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/start/do-it-for-me` and go through the flow steps.
* On the page picker step, hover over the info icon for each page and confirm that the page description is displayed.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes Automattic/martech#1124
Fixes Automattic/martech#1142

Commit:
DIFM: Add description as tooltip on page picker step

Commit:
Adds popular badge to a few pages

ddc22 commented on client/signup/difm/translation-hooks.tsx:
```diff
@@ -50,3 +50,49 @@ export function useTranslatedPageTitles() {
+
+export function useTranslatedPageDescriptions() {
+	const translate = useTranslate();
+	const descriptions: Record< string, TranslateResult > = useMemo(
```
Not sure we need to optimize this since `translate()` is already heavily optimised [using an LRU cache](https://github.com/Automattic/wp-calypso/blob/54ff4556c2e4eeae2810d0e93ae212cb4e3b0731/packages/i18n-calypso/src/i18n.js#L148). Having these values in a JS object is already a hash map and so we can simply return the plain object. Feel like this might be a bit of [unnecessary complexity](https://kentcdodds.com/blog/usememo-and-usecallback).

aneeshd16 replied:
TIl about the LRU cache! I followed the same pattern as `useTranslatedPageTitles` in the same file. But if the cache is in place, then we don't need useMemo in both hooks.

ddc22 reviewed (approved):
Minor z-index bug which might have to be fixed on mobile as shown below and a minor code feedback. But other than that LGTM. 🚢 
<img width="456" alt="Screenshot 2022-10-13 at 1 11 57 PM" src="https://user-images.githubusercontent.com/3422709/195555526-a1b02e63-5453-48ef-877d-12341a8316aa.png">

Commit:
Update copy and fixes z-index issue

Commit:
Removed redundant useMemo

aneeshd16 commented:
> Minor z-index bug which might have to be fixed on mobile as shown below and a minor code feedback. But other than that LGTM. 🚢 

Thanks, I have fixed the z-index issue as well as addressed the code feedback.

Commit:
Update copy

ddc22 reviewed (approved):
Feel like the tooltip could tilt more to the left a bit more so that the fact that it belongs to the relevant page type is more clear with closer proximity. But that can be looked at later if required. 
<img width="990" alt="image" src="https://user-images.githubusercontent.com/3422709/196118899-76eaf256-1f4d-456c-9261-081a3bb86f6a.png">


For now the feature works well 💯 

LGTM 🚢
